pkg/aws/client.go

```diff
@@ -1065,28 +1008,29 @@ const ReadOnlyAnonUserPolicyTemplate = `{
 }`
 
 func (c *awsClient) CreateS3Bucket(bucketName string, region string) error {
-	_, err := c.s3Client.HeadBucket(&s3.HeadBucketInput{
+	_, err := c.s3Client.HeadBucket(context.TODO(), &s3.HeadBucketInput{
 		Bucket: aws.String(bucketName),
 	})
 	if err == nil {
 		return weberr.Errorf("Bucket '%s' already exists.", bucketName)
 	}
+
 	bucketInput := &s3.CreateBucketInput{
 		Bucket: aws.String(bucketName),
 	}
 	if region != DefaultRegion {
-		bucketInput.SetCreateBucketConfiguration(&s3.CreateBucketConfiguration{
-			LocationConstraint: &region,
-		})
+		bucketInput.CreateBucketConfiguration = &s3types.CreateBucketConfiguration{
+			LocationConstraint: s3types.BucketLocationConstraint(region),
+		}
 	}
-	_, err = c.s3Client.CreateBucket(bucketInput)
+	_, err = c.s3Client.CreateBucket(context.TODO(), bucketInput)
```

Comment:
There is a mixture of context.TODO() and context.Background() is this something we want to do or should we be creating a context at a higher level and passing it down :thinking:

Reply:
Has this been decided to been done as part of a separate ticket?